pkg/cloudprovider/providers/aws/aws_test.go

```diff
@@ -609,7 +612,18 @@ func TestNodeAddresses(t *testing.T) {
 	}
 	instance1.State = &state1
 
-	instances := []*ec2.Instance{&instance0, &instance1}
+	//2
+	instance2.InstanceId = aws.String("i-self")
```

Comment:
Had to add this new instance because of the way the tests are setup so that it reaches the code in `NodeAddresses` when the instance name that is searched is different from the current instance node name.

The `NodeAddresses` tries to do too much imho and has a lot of dead code which made updating the tests much more complicated than it should be. I'll open a different issue for that but I think we should refactor `NodeAddresses` on the model of the `gce` one. I didn't do the refactoring here because I didn't want to "muddy" the PR.